keps/sig-storage/20200508-move-utils-mount-2staging.md

```diff
@@ -0,0 +1,108 @@
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+  - [Version Skew Strategy](#version-skew-strategy)
+- [Implementation History](#implementation-history)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [X] KEP approvers have set the KEP status to `implementable`
+- [NA] Design details are appropriately documented
+- [X] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [NA] Graduation criteria is in place
+- [X] "Implementation History" section is up-to-date for milestone
+- [NA] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [NA] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+
+## Summary
+
+We want to move vendor/utils/mount library into k/k staging to better test library fixes and manage its release process more efficiently.
+Currently utils/mount is vendor in and used by CSI driver framework. Whenever a fix is issued to the library it will not be possible to test the core CSI code to validate the fix. We would like to run the e2e test to really make sure the fix works. Also it is hard to backport some of these potential fixes to previous Kubernetes releases. This code really should belong to core Kubernetes to better manage.   
+
+## Motivation
+
+The main context is that the mount library is useful for csi drivers. Main motivations are: 
+1. To make our e2e tests exercise the changes made to utils/mount during prs, because there's an additional step that they have to get vendored into k8s
+2. There's no easy way to backport mount fixes to older k8s releases without potentially pulling in a number of unrelated libraries in k8s/utils
+
+
+### Goals
+
+To move the vendored utils/mount library into kubernetes. Avoid the above mentioned issues with vendored code.
+
+### Non-Goals
+
+
+
+## Proposal
+
+Move the code and maintain under /staging/src/k8s.io/utils/mount. Similar to csi-translation-lib. 
+
+### Risks and Mitigations
+
+None, as the code should just compile and work fine from the new location.
+
+## Design Details
+
+### Test Plan
+
+All existing e2e tests and unit test should work fine and will go through normal PR testing.
+
+### Graduation Criteria
+
+None
+
+### Upgrade / Downgrade Strategy
+
+No upgrade/downgrade concerns.
+
+### Version Skew Strategy
+
+None
+
+## Implementation History
+
+2020-05-08: KEP opened
+2020-05-08: KEP marked implementable
+
+## Alternatives
+
+We do not see better alternative to maintain this code external to k/k as this is part of code maintained by sig-storage and used in core CSI code.
```

Comment:
Another alternative is to create a new separate repo under k8s/.  But the main downsides are that we would still need to invest heavily to build and maintain tests and infrastructure with the same coverage that we have in k/k.

Reply:
Done